src/libcore/ops.rs

```diff
@@ -1530,6 +1533,75 @@ impl<Idx: fmt::Debug> fmt::Debug for RangeTo<Idx> {
     }
 }
 
+/// An inclusive range which is bounded at both ends.
+#[derive(Copy, Clone, PartialEq, Eq)]
+#[unstable(feature = "inclusive_range", reason = "recently added, follows RFC", issue = "28237")]
+pub enum RangeInclusive<Idx> {
+    /// Empty range (iteration has finished)
+    Empty {
+        /// The point at which iteration finished
+        at: Idx
+    },
+    /// Non-empty range (iteration will yield value(s))
+    NonEmpty {
```

Comment:
Yes, `5...5` results in a lie. However, `7..3` is also a lie in some sense. And a point of consolation is that if you actually [try to use the lie as a range](https://github.com/rust-lang/rust/pull/30884/files#diff-5e4215f9d47e2b6d5fb23cc8dd1d79d8R102), it returns the correct result (nothing) and modifies itself to not be a lie :)